Release RC2 (#271)
JoshOrndorff/recipes: text/3-entrees/currency.md

# Currency Types

_[`pallets/lockable-currency`](https://github.com/substrate-developer-hub/recipes/tree/master/pallets/lockable-currency),
[`pallets/reservable-currency`](https://github.com/substrate-developer-hub/recipes/tree/master/pallets/reservable-currency),
[`pallets/currency-imbalances`](https://github.com/substrate-developer-hub/recipes/tree/master/pallets/currency-imbalances)_

## Just Plain Currency

> To use a balances type in the runtime, import the

Reviewer: ```suggestion
To use a balance type in the runtime, import the
```
?

[`Currency`](https://substrate.dev/rustdocs/v2.0.0-rc2/frame_support/traits/trait.Currency.html) trait from
`frame_support`.

```rust, ignore
use support::traits::Currency;
```

> The `Currency` trait provides an abstraction over a fungible assets system. To use such a fuingible

Reviewer: ```suggestion
The `Currency` trait provides an abstraction over a fungible assets system. To use such a fungible
```

Also: I think it could be useful to have link to an explanation of what a "fungible asset" is, maybe just https://en.wikipedia.org/wiki/Fungibility

Author: ```suggestion
The `Currency` trait provides an abstraction over a [fungible assets](https://en.wikipedia.org/wiki/Fungibility) system. To use such a fungible
```

type BalanceOf<T> = <<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::Balance;
```

This new `BalanceOf<T>` type satisfies the type constraints of `Self::Balance` for the provided
methods of `Currency`. This means that this type can be used for
[transfer](https://substrate.dev/rustdocs/v2.0.0-rc2/frame_support/traits/trait.Currency.html#tymethod.transfer),
[minting](https://substrate.dev/rustdocs/v2.0.0-rc2/frame_support/traits/trait.Currency.html#tymethod.deposit_into_existing),
and much more.

## Reservable Currency

Substrate's [Treasury pallet](https://substrate.dev/rustdocs/v2.0.0-rc2/pallet_treasury/index.html) uses the

> `Currency` type for bonding spending proposals. To reserve and unreserve balances for bonding,

Reviewer: ```suggestion
`Currency` type for bonding spending proposals. To reserve and unreserve funds for bonding,
```
maybe?
